infra/scripts/local-ci.sh

```diff
@@ -0,0 +1,198 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##### defaults
+secretsfile=""
+docker_test="Group1-Docker-Commands"
+target_vch=""
+odir="ci-results"
+ci_container="gcr.io/eminent-nation-87317/vic-integration-test:1.39"
+github_api_key=""
+test_url=""
+test_datastore=""
+test_username=""
+test_password=""
+vic_dir=$(pwd)
+
+##### utility functions
+function usage() {
+  echo "Usage: $0 [options]" 1>&2
+  echo
+  echo "  Options can be provided by commandline argument, environment variable, or a secrets yaml file containing the"
+  echo "  variables.  If a secrets file is not provided, this script will attempt to retrieve some info from govc, such as"
+  echo "  TEST_URL, TEST_USERNAME, and TEST_PASSWORD."
+  echo
+  echo "  options:"
+  echo "    -t DOCKER_TEST (or env var)"
+  echo "    -v TARGET_VCH (or env var)"
+  echo "    -s SECRETS_FILE (or env var)"
+  echo "    -g GITHUB_API_KEY (or env var)"
+  echo "    -u TEST_URL (or env var or in secretsfile)"
+  echo "    -d TEST_DATASTORE (or env var or in secretsfile)"
+  echo "    -n TEST_USERNAME (or env var or in secretsfile)"
+  echo "    -p TEST_PASSWORD (or env var or in secretsfile)"
+  echo
+  echo "  example:"
+  echo "    $0 -t Group1-Docker-Commands -v my_vch -s test.secrets.nested -g xxxxxx"
+  echo "    $0 -t 1-01-Docker-Info.robot -v my_vch -s test.secrets.nested -g xxxxxx"
+  echo 
+  echo "    DOCKER_TEST=Group1-Docker-Commands/1-01-Docker-Info.robot TARGET_VCH=my_vch SECRETS_FILE=test.secrets.nested $0"
+  echo
+  echo "    $0 -s test.secrets.nested (all params defined in secrets file)"
+  exit 1
+}
+
+function GetGovcParamsFromEnv() {
+  echo "Getting params from GOVC var"
+  test_username=$(govc env | grep GOVC_USERNAME | cut -d= -f2)
+  test_password=$(govc env | grep GOVC_PASSWORD | cut -d= -f2)
+  test_url=$(govc env | grep GOVC_URL | cut -d= -f2)
+}
+
+function GetParamsFromSecrets() {
+  echo "Getting params from the secrets file"
+  secrets_api_key="$(grep 'GITHUB_AUTOMATION_API_KEY' ${secretsfile} | awk '{ print $2 }')"
+  secrets_url="$(grep 'TEST_URL_ARRAY' ${secretsfile} | awk '{ print $2 }')"
+  secrets_datastore="$(grep -E '\s+TEST_DATASTORE' ${secretsfile} | awk '{ print $2 }')"
+  secrets_username="$(grep 'TEST_USERNAME' ${secretsfile} | awk '{ print $2 }')"
+  secrets_password="$(grep 'TEST_PASSWORD' ${secretsfile} | awk '{ print $2 }')"
+  secrets_docker_test="$(grep 'DOCKER_TEST' ${secretsfile} | awk '{ print $2 }')"
+  secrets_target_vch="$(grep 'TARGET_VCH' ${secretsfile} | awk '{ print $2 }')"
+}
+
+##### Get command line arguments
+while getopts s:t:v:g:u:d:n:p: flag
+do
+  case $flag in
+    s)
+      secretsfile=$OPTARG
+      ;;
+    t)
```

Comment:
I think you can provide `pybot -t` to run only a specific test in a suite. We may want to use a different option here so adding pass-through for that later doesn't result in option name collision.